src/discovery/discovery.utils.ts

```diff
@@ -0,0 +1,314 @@
+import { constants, getDefaultProvider, providers } from 'ethers';
+import {
+  getAddress,
+  HDNode,
+  keccak256,
+  solidityKeccak256,
+} from 'ethers/lib/utils';
+
+import { langCz as cz } from '@ethersproject/wordlists/lib/lang-cz';
+import { langEn as en } from '@ethersproject/wordlists/lib/lang-en';
+import { langEs as es } from '@ethersproject/wordlists/lib/lang-es';
+import { langFr as fr } from '@ethersproject/wordlists/lib/lang-fr';
+import { langIt as it } from '@ethersproject/wordlists/lib/lang-it';
+import { langJa as ja } from '@ethersproject/wordlists/lib/lang-ja';
+import { langKo as ko } from '@ethersproject/wordlists/lib/lang-ko';
+import {
+  langZhCn as zh_cn,
+  langZhTw as zh_tw,
+} from '@ethersproject/wordlists/lib/lang-zh';
+import type { SmartWalletFactory } from '@rsksmart/rif-relay-contracts/dist/typechain-types/contracts/factory/SmartWalletFactory';
+import { ERC20__factory } from '@rsksmart/rif-relay-contracts/dist/typechain-types/factories/@openzeppelin/contracts/token/ERC20/ERC20__factory';
+import { SmartWalletFactory__factory } from '@rsksmart/rif-relay-contracts/dist/typechain-types/factories/contracts/factory';
+const SHA3_NULL_S =
+  '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470'; //TODO: verify this is the correct way to get the hash. Copied from rif-relay-common (https://github.com/rsksmart/rif-relay-common/blob/f8e4ce8d90a979f3eec55a0eb5dbd5cc742c40e3/src/Constants.ts#L18-L19)
```

Comment:
If you apply a keccak-256 to an empty string, this is the value that it returns. Im not sure if need to have in a constant or generate dynamically.

Reply:
aaaaaah right... thanks!

Reply:
done